### PR TITLE
symintify nll loss fns

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86915

### CODE REVIEW

**Review by albanD — APPROVED** [review]
SGTM!
I'm sure some xfail will need updating?

**Comment by anjali411** [comment]
hmm I don't see any related test failures (https://github.com/pytorch/pytorch/pull/86606 also symintified nll loss in the past and didn't need any updates)

File: torch/csrc/jit/runtime/static/generated_ops.cpp [hunk 1]
```diff
@@ -3437,13 +3437,12 @@ REGISTER_OPERATOR_FUNCTOR(aten::nll_loss, aten_nll_loss, [](Node* n) -> SROperat
       const auto ignore_index = p_node->Input(4).toInt();
       if (p_node->Output(0).isNone()) {
         p_node->Output(0) =
-            at::native::nll_loss(self, target, weight, reduction, ignore_index);
+            at::nll_loss(self, target, weight, reduction, ignore_index);
```
Comment (ezyang) [inline torch/csrc/jit/runtime/static/generated_ops.cpp:3440]: This is a pessimization for static runtime. Should apply the BC trick so the old at::native:: is available

File: torch/csrc/jit/runtime/static/generated_ops.cpp [hunk 2]
```diff
@@ -3465,7 +3464,7 @@ REGISTER_OPERATOR_FUNCTOR(
           const auto ignore_index = p_node->Input(5).toInt();
           const auto& total_weight = p_node->Input(6).toTensor();
           if (p_node->Output(0).isNone()) {
-            p_node->Output(0) = at::cpu::nll_loss_backward(
+            p_node->Output(0) = at::nll_loss_backward(
```
Comment (ezyang) [inline torch/csrc/jit/runtime/static/generated_ops.cpp:3467]: this change is completely unnecessary, I think the old change should have compiled

File: torch/csrc/jit/runtime/static/generated_ops.cpp (hunk 1 above)
Comment (anjali411) [reply torch/csrc/jit/runtime/static/generated_ops.cpp:3440]: what's the BC trick?

Comment (ezyang) [reply torch/csrc/jit/runtime/static/generated_ops.cpp:3440]: Put the old signature in aten/src/ATen/native/NonSymbolicBC.h and then template the native:: declaration to define both the non-symint and symint simultaneously

Comment (anjali411) [reply torch/csrc/jit/runtime/static/generated_ops.cpp:3440]: oh nice. addressed in the reland PR https://github.com/pytorch/pytorch/pull/87095

**Pull request closed without merging**